### PR TITLE
WIP: Add support for different versions of Artifactory

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Different versions allow for different features.
  This adds support for the enterprise version (ent)
  which allows you to attach buildinfo to the artifacts.
  This also adds support for the OpenSource version (oss)
  which does not allow buildinfo.

Issue #2637
Related AdoptOpenJDK/openjdk-tests#898
[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Test build
https://ci.eclipse.org/openj9/view/Sandbox/job/Adam-Pipeline-Build-Test-All/12/
@jdekonin for review
Note: This doesn't enable Artifactory on Eclipse Jenkins. It only adds support for various versions. See #4705 for enablement.

File: buildenv/jenkins/common/build [hunk 1]
```diff
@@ -324,13 +324,22 @@ def archive() {
...
+                        // Add BUILD_IDENTIFIER to the buildInfo. The UploadSpec adds it to the Artifact info
+                        buildInfo.env.filter.addInclude("BUILD_IDENTIFIER")
+                        buildInfo.env.capture = true
+                        server.upload spec: uploadSpec, buildInfo: buildInfo
+                        server.publishBuildInfo buildInfo
+                    } else if (ARTIFACTORY_VERSION == "oss") {
+                        echo "Not adding Artifactory buildInfo, unsupported on OSS Version"
+                        server.upload spec: uploadSpec
+                    } else {
+                        error "Unknown Artifacotry version:${ARTIFACTORY_VERSION}"
```
Comment (jdekonin) [inline buildenv/jenkins/common/build]: Spelling of `Artifacotry`.

**Comment by AdamBrousseau** [comment]
Spelling fixed.
@smlambert or @llxia for merge.

**Comment by AdamBrousseau** [comment]
Retagged as WIP while I work on getting the version from the REST API.

**Comment by AdamBrousseau** [comment]
close in favour of #5179

**Pull request closed without merging**